test/metabase/query_processor_test/expressions_test.clj

```diff
@@ -184,3 +191,54 @@
 (datasets/expect-with-drivers (qp.test/non-timeseries-drivers-with-feature :expressions)
   [[nil] [0.0] [0.0] [10.0] [8.0] [5.0] [5.0] [nil] [0.0] [0.0]]
   (calculate-bird-scarcity [:* 1 [:field-id $count]]))
+
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                           DATETIME EXTRACTION AND MANIPULATION                                           |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(def ^:private utc-tz (time/time-zone-for-id "UTC"))
+
+(defn- maybe-truncate
+  [dt]
+  (if (= :sqlite driver/*driver*)
+    (->> dt (du/date-trunc :day) tcoerce/from-sql-date)
+    dt))
+
+(defn- robust-dates
+  [dates]
+  (let [output-format (if (= :sqlite driver/*driver*)
+                        :mysql
+                        :date-time)]
+    (for [d dates]
+      [(->> d
+            (tformat/parse (tformat/with-zone (tformat/formatters :date-hour-minute-second-fraction) utc-tz))
+            maybe-truncate
+            (tformat/unparse (tformat/with-zone (tformat/formatters output-format) utc-tz)))])))
+
+(datasets/expect-with-drivers (qp.test/non-timeseries-drivers-with-feature :expressions)
```

Comment:
can you write a comment above the test explaining what it's testing? e.g. something like

```clj
;; test that we can use MBQL `:interval` clauses in expressions
```

When we come back a year later it's a lot simpler to read the comment explaining the test than to have to read the whole test and work out what's being tested